src/Core/Managed/Net40/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -44,8 +44,27 @@ protected TelemetryConfigurationFactory()
         /// </remarks>
         public static TelemetryConfigurationFactory Instance
         {
-            get { return instance ?? (instance = new TelemetryConfigurationFactory()); }
-            set { instance = value; }
+            // Both get and set should write something to CoreEventSource so as to initialize it. This is a fix for below deadlock issue   
```

Comment:
The original issue is in TraceSource in Azure Web Sites trace listener. So strictly speaking this change is not necessary - you can modify Azure Web Site trace listener as recommended by CLR(BCL) team.